### PR TITLE
Handle attribute parameters when converting ONNX to ONNXScript

### PR DESCRIPTION
**Pull request by gramalingam**
The ONNX to ONNXScript exporter does not handle attribute parameters of functions. Add this missing functionality.

### CODE REVIEW

File: onnxscript/backend/onnx_export.py [hunk 1]
```diff
@@ -26,17 +27,17 @@
 {% for domain, version in unique_function_domain_version: %}
 {{ domain }}{{ version }} = Opset("{{ domain }}", {{ version }}){% endfor %}
 {% for domain, name, fct in functions: %}
-@script({{ domain }}{{ version }})
-def {{ python_make_node_name(fct['proto'].domain, 1, fct['proto'].name) }}({{
-    ", ".join(map(rename, fct['proto'].input)) }}):
-    # attributes are missing
+@script({{ domain }}1)
```
Comment (justinchuby) [inline onnxscript/backend/onnx_export.py:30]: Was the 1 intended?

File: onnxscript/backend/onnx_export.py [hunk 2]
```diff
@@ -419,6 +428,51 @@ def _python_make_node(self, onnx_node, opsets, indent=0):
...
+        def attr_sig(attr_name: str) -> str:
+            name = self._rename_variable(attr_name)
+            # A default type of INT is used for attribute parameters that are never used.
+            type = type_map.get(attr_name, onnx.AttributeProto.INT)
+            typerep = onnxscript.type_annotation.onnx_attr_type_to_onnxscript_repr(type)
+            return f"{name}: {typerep}"
+
+        inputs = [self._rename_variable(x) for x in funproto.input]
+        attrs = [attr_sig(x) for x in funproto.attribute]
+        input_and_attrs = ", ".join(inputs + attrs)
```
Comment (justinchuby) [inline onnxscript/backend/onnx_export.py:444]:
```suggestion
        input_and_attrs = ", ".join((*inputs, *attrs))
```

File: onnxscript/type_annotation.py [hunk 3]
```diff
@@ -43,6 +43,23 @@
...
+    onnx.AttributeProto.STRING: "str",
+    onnx.AttributeProto.FLOATS: "Sequence[float]",
+    onnx.AttributeProto.INTS: "Sequence[int]",
+    onnx.AttributeProto.STRINGS: "Sequence[str]",
+}
+
+
+def onnx_attr_type_to_onnxscript_repr(attr_type: onnx.AttributeProto.AttributeType) -> str:
+    if attr_type not in _ATTRTYPE_TO_REPR:
+        raise ValueError("Unsupported attribute type {attr_type}")
```
Comment (justinchuby) [inline onnxscript/type_annotation.py:59]: Include the supported in the error message?

File: onnxscript/backend/onnx_export.py (hunk 1 above)
Comment (gramalingam) [reply onnxscript/backend/onnx_export.py:30]: Yes, that's what the rest of the code-generator uses.

File: onnxscript/backend/onnx_export.py (hunk 2 above)
Comment (gramalingam) [reply onnxscript/backend/onnx_export.py:444]: I find this harder to understand: it is a more roundabout way to concatenate two lists.

Comment (gramalingam) [reply onnxscript/backend/onnx_export.py:444]: The warning message earlier should be addressed some other way. Will take a look at it.

File: onnxscript/type_annotation.py (hunk 3 above)
Comment (gramalingam) [reply onnxscript/type_annotation.py]: Done

**Review by justinchuby — APPROVED** [review]

**Pull request merged**